src/App.tsx

```diff
@@ -2,10 +2,46 @@ import React from 'react'
 import logo from './logo.svg'
 import './App.css'
 import FileImportButton from './Components/FileImportButton'
+import TextField from '@material-ui/core/TextField'
+import xml2js from 'xml2js'
+import Attempt from './types'
 
 const App = () => {
-  function handleFile() {
-    console.log('test')
+  const [attemptHistory, setAttemptHistory] = React.useState<Attempt[]>([])
+
+  function handleFile(e: React.ChangeEvent<HTMLInputElement>) {
+    if (e?.target?.files?.[0]) {
+      const file = e.target.files[0]
+
+      const reader = new FileReader()
+      reader.onload = () => {
+        const xml = reader.result?.toString()
+
+        if (xml) {
+          xml2js.parseString(xml, { trim: true }, async function(err, result) {
+            //todo: display error nicely in UI
+            if (err) {
+              console.log(err)
+            }
+
+            await getAttemptHistory(result)
+          })
+        }
+      }
+      reader.readAsText(file)
+    }
+  }
+
+  async function getAttemptHistory(xml: any) {
+    let atmptHistory: Attempt[] = []
+
+    xml?.Run?.AttemptHistory?.[0]?.Attempt?.forEach((attempt: any) => {
+      if (attempt?.$?.id && attempt?.RealTime?.[0]) {
+        atmptHistory.push({ x: attempt.$.id, y: attempt.RealTime[0] })
+      }
+    })
```

Comment:
This is technically fine.
An alternative could also be:
```
const attemptHistory: Attempt[] = xml?.Run?.AttemptHistory?.[0]?.Attempt?.flatMap(({ $: any, RealTime: any }: any) => {
  return $?.id && RealTime?.[0] ? [{ x: $.id, y: RealTime.[0] }] : []
})
```
(I just guessing at the typing of `({ $: any, RealTime: any }: any)` but `({$, Realtime})` is valid JS.
I also wonder if you can guarantee some more of those values to not be nullable given the structure of the input?

Reply:
I still need to look into xml2js some more to see what all my options are.  It looks like I may be able to read directly from a file removing the need for FileReader all together.

Reply:
Also made the suggested change